homeassistant/components/filter/sensor.py

```diff
@@ -279,8 +285,10 @@ async def async_added_to_hass(self):
             for state in history_list:
                 self._update_filter_sensor_state(state, False)
 
-        async_track_state_change_event(
-            self.hass, [self._entity], self._update_filter_sensor_state_event
+        self.async_on_remove(
```

Comment:
What does this do ?

Reply:
It unsubscribes the listener when the entity it removed so it doesn't keep calling back to an entity that is no longer there. 

Some of the older integrations are missing these since they never expected the entity could be removed once it was created.

Reply:
tks for the insight!